src/scan-bcl/Program.cs

```diff
@@ -0,0 +1,176 @@
+                throw new DirectoryNotFoundException($"BCL Path not found at : {fiveBCLPath}");
+            if (!Directory.Exists(outputDirPath)) {
+                Directory.CreateDirectory(outputDirPath);
+            }
+
+            var threeOutputPath = Path.Combine(outputDirPath, "three-bcl.csv");
+            var threeScan = ScanBCL (scanPath, threeBCLPath, threeOutputPath);
+
+            var fiveOutputPath = Path.Combine(outputDirPath, "five-bcl.csv");
+            var fiveScan = ScanBCL (scanPath, fiveBCLPath, fiveOutputPath);
+
+            if (threeScan && fiveScan) {
+                var diff = Scan (threeOutputPath, fiveOutputPath);
+                var outputPath = Path.Combine(outputDirPath, "diff.csv");
+                ExportCsv(diff, outputPath);
+            }
+        }
+
+        private static bool ScanBCL (string scanPath, string BCLPath, string outputPath) {
+            var command = "mono";
+            var args = $@"{scanPath} {BCLPath} {outputPath}";
+
+            var process = new Process();
+            process.StartInfo.FileName = command;
+            process.StartInfo.Arguments = args;
+            process.StartInfo.RedirectStandardError = true;
+            process.StartInfo.RedirectStandardOutput = true;
+            process.StartInfo.UseShellExecute = false;
+            process.OutputDataReceived += (s, d) => Console.Out.WriteLine(d.Data);
+            process.ErrorDataReceived += (s, d) => Console.Error.WriteLine(d.Data);
+            process.Start();
+            process.BeginOutputReadLine();
+            process.BeginErrorReadLine();
+            process.WaitForExit();
+            return process.ExitCode == 0;
+        }
+
+        private static Database Scan (string threeOutputPath, string fiveOutputPath) {
+            Console.WriteLine("Analyzing...");
+
+            Database threeBCLDatabase = null;
+            var diff = new Database ();
+            if (threeOutputPath != null && fiveOutputPath != null) {
+                threeBCLDatabase = new Database ();
+                ImportCsv(threeOutputPath, threeBCLDatabase);
+                ImportCsv(fiveOutputPath, threeBCLDatabase);
+
+                diff = threeBCLDatabase;
+            }
+            return diff;
+        }
+
+        private static void ImportCsv(string path, Database database)
+        {
+             
+            using (var streamReader = new StreamReader(path))
+            using (var csvReader = new CsvReader(streamReader, CultureInfo.InvariantCulture))
+            {
+                // csv.Configuration.RegisterClassMap<DatabaseMap>();
+                csvReader.Read();
+                var header = csvReader.ReadHeader();
+                while (csvReader.Read()) {
+                    var row = csvReader.GetRecord<DatabaseEntry>();
+                    DatabaseEntry query = null;
+                    try{
+                        query = database.Entries.Single(x => row.DocId == x.DocId && 
```

Comment:
I think the DocId is unique enough to compare against which allows you to use the key to lookup the value in the table which should be much quicker.   But did you consider using two databases (or adding more fields) to keep track of which api group it existed in?

Reply:
I will edit ex-scan so that it also includes the api.